src/functions/topsort-client.ts

```diff
@@ -0,0 +1,42 @@
+import { baseURL } from "../constants/apis.constant";
+import AppError from "../lib/app-error";
+import {
+  AuctionResult,
+  EventResult,
+  TopsortAuction,
+  TopsortEvent,
+} from "../types";
+import { Config } from "../types/shared";
+
+import { createAuction } from "./auctions";
+import { reportEvent } from "./events";
+
+class TopsortClient {
+  private config: Config;
+
+  constructor(config: Config) {
+    this.config = config;
+    this.config.host = this.sanitizeUrl(this.config.host ?? baseURL);
+  }
+
+  private sanitizeUrl(url: string): string {
+    try {
+      const parsedUrl = new URL(url);
+      return parsedUrl.href.replace(/\/+$/, "");
```

Comment:
I don't think it makes sense to do this until we merge it with path in the future

Reply:
Yep, I just changed the function, will push in a bit